### DISCUSSION TITLE
bat is launching vim when I try to bat a .sh file

### DISCUSSION THREAD

**Discussion by twiclo** [sharkdp/bat#3141] [Q&A]
```
[tanner@sato api]$ ^C
[tanner@sato api]$ /usr/bin/cat test.sh
#!/bin/bash

export $(grep -v '^#' .env | xargs)
NODE_OPTIONS="--preserve-symlinks" ENV=TEST npm run test
[tanner@sato api]$ /usr/bin/bat test.sh 
Vim: Warning: Input is not from a terminal
Vim: Error reading input, exiting...
Vim: preserving files...
Vim: Finished.
[tanner@sato api]$ 1u
```

Is this a bug or is there a common setting I'm missing?

**Comment by twiclo** [comment]
Nevermind. I had vim set as my PAGER

**Comment by fotedev** [comment]
Not a bug - it is the pager selection at work. When bat pages, it uses the `--pager` flag first, then `$BAT_PAGER`, then `$PAGER`, falling back to `less`. With `PAGER=vim` exported, bat dutifully pipes the file into vim, and vim complains because its stdin is a pipe rather than a terminal ("Input is not from a terminal").

Fix, pick one:

```sh
export BAT_PAGER="less -R"   # bat-specific, leaves PAGER alone
unset PAGER                  # or: export PAGER=less
```

`BAT_PAGER` is the cleanest option since it only affects bat.
